agent_api/src/main/java/dev/aikido/agent_api/collectors/WebRequestCollector.java

```diff
@@ -19,20 +19,25 @@ public record Res(String msg, Integer status) {};
      * This function gets called in the initial phases of a request.
      * @param newContext is the new ContextObject that holds headers, query, ...
      * Also returns a response depending on blocking status :
-     * 1) First check: IP allowed to access route (restricted /admin routes e.g.)
-     * 2) Bypassed IP? If bypassed, no further checks
+     * 1) If the IP is bypassed we just won't set the context object
+     * 2) IP allowed to access route (restricted /admin routes e.g.)
      * 3) Blocked IP Lists: e.g. geo-blocking, crowdsec, ...
      * 4) UA Blocking: e.g. bot blocking, AI scraper blocking, ...
      */
     public static Res report(ContextObject newContext) {
+        ThreadCacheObject threadCache = ThreadCache.get();
         // Set new context :
         Context.reset();
-        Context.set(newContext);
-        ThreadCacheObject threadCache = ThreadCache.get();
+
+        if (threadCache != null && threadCache.isBypassedIP(newContext.getRemoteAddress())) {
+            return null; // Do not set context if IP is bypassed.
+        } else {
+            Context.set(newContext);
+        }
```

Comment:
```suggestion
        }

        Context.set(newContext);
```

Maybe remove the else, that reads a bit nicer I think